hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -163,6 +169,108 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     }
   }
 
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> meta) {
+    Preconditions.checkArgument(!namespace.isEmpty(),
+        "Cannot create namespace with invalid name: %s", namespace.toString());
+    Preconditions.checkArgument(namespace.levels().length == 1,
+        "Hive MetaStore cannot support multi part namespace now: %s", namespace.toString());
+    try {
+      clients.run(client -> {
+        client.createDatabase(nameSpaceToHiveDb(namespace, meta));
+        return null;
+      });
+
+    } catch (AlreadyExistsException e) {
+      throw new org.apache.iceberg.exceptions.AlreadyExistsException("namespace already exists: %s",
+            namespace.toString());
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to create namespace " + namespace.toString() + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to createDatabase(name)", e);
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    if (!namespace.isEmpty()) {
+      Preconditions.checkArgument(namespace.levels().length == 1,
+          "Hive MetaStore cannot support multi part namespace now: %s", namespace.toString());
+    }
+    List<Namespace> namespaces = new ArrayList<>();
+    List<String> dbs;
+    try {
+      dbs = namespace.isEmpty() ? clients.run(HiveMetaStoreClient::getAllDatabases) :
+           clients.run(client -> client.getDatabases(namespace.level(0)));
+      for (String db : dbs) {
+        namespaces.add(Namespace.of(db));
+      }
+
+      return namespaces;
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list all namespace: " + e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to getAllDatabases()", e);
+    }
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    Preconditions.checkArgument(!namespace.isEmpty(),
+        "Namespace could not empty. ");
+    Preconditions.checkArgument(namespace.levels().length == 1,
+        "Hive MetaStore cannot support multi part namespace now: %s", namespace.toString());
```

Comment:
If there is more than one part, the namespace can't exist. Instead, throw NoSuchNamespaceException.